.github/workflows/build-and-release-edge.yml

```diff
@@ -120,22 +111,40 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
-          ref: ${{ inputs.branch }} 
+          ref: ${{ inputs.branch }}
       - name: Docker login
         uses: azure/docker-login@v1
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       - name: Docker edge build & tag
         run: |
-          DOCKER_BUILDKIT=1 docker build -f install/docker/Dockerfile --no-cache -t ${{ secrets.IMAGE_NAME }}:edge-latest --build-arg TOKEN=${{ secrets.GLOBAL_TOKEN }} --build-arg GIT_COMMITSHA=${GITHUB_SHA::8} --build-arg GIT_VERSION=${{ env.LATEST_TAG }} --build-arg RELEASE_CHANNEL=${RELEASE_CHANNEL} .
+          DOCKER_BUILDKIT=1 docker build -f install/docker/Dockerfile --no-cache \
+            -t ${{ secrets.IMAGE_NAME }}:edge-latest \
+            --build-arg TOKEN=${{ secrets.GLOBAL_TOKEN }} \
+            --build-arg GIT_COMMITSHA=${GITHUB_SHA::8} \
+            --build-arg GIT_VERSION=${{ env.LATEST_TAG }} \
+            --build-arg RELEASE_CHANNEL=${RELEASE_CHANNEL} .
           docker tag ${{ secrets.IMAGE_NAME }}:edge-latest ${{ secrets.IMAGE_NAME }}:edge-${GITHUB_SHA::8}
           docker tag ${{ secrets.IMAGE_NAME }}:edge-latest ${{ secrets.IMAGE_NAME }}:edge-${{ env.LATEST_TAG }}
       - name: Docker edge push
         run: |
           docker push ${{ secrets.IMAGE_NAME }}:edge-latest
           docker push ${{ secrets.IMAGE_NAME }}:edge-${GITHUB_SHA::8}
           docker push ${{ secrets.IMAGE_NAME }}:edge-${{ env.LATEST_TAG }}
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@v3.5.0
+      - name: Sign Docker images (keyless)
+        run: |
+          for tag in edge-latest edge-${GITHUB_SHA::8} edge-${{ env.LATEST_TAG }}; do
+            echo "Signing ${{ secrets.IMAGE_NAME }}:$tag"
+            cosign sign docker.io/${{ secrets.IMAGE_NAME }}:$tag --keyless
+          done
+      - name: Verify Docker image signatures
+        run: |
+          for tag in edge-latest edge-${GITHUB_SHA::8} edge-${{ env.LATEST_TAG }}; do
+            echo "Verifying ${{ secrets.IMAGE_NAME }}:$tag"
+            cosign verify docker.io/${{ secrets.IMAGE_NAME }}:$tag --keyless
 
```

Comment:
We likely need this scoped to meshery somehow? As im wondering if this verification would allow someone else to essentially sign it. 

https://docs.sigstore.dev/cosign/verifying/verify/#keyless-verification-using-openid-connect.